guppylang/checker/core.py

```diff
@@ -162,41 +162,45 @@ def __getitem__(self, item: DefId | str) -> Definition:
                 return assert_never(x)
 
 
+K = TypeVar("K")
+V = TypeVar("V")
```

Comment:
We actually have `K == P`

During type checking, we have `K = P = str` and `V = Variable`. During linearity checking we have `K = P = PlaceId` and `V = Place` (see #290).

The idea is that `K` and `P` are identifiers for program variables that can be used to lookup a `V` which holds extra data associated with the program variables (e.g. their type, definition location, etc.)

Maybe we should rename `K, P -> VId`??

Reply:
Ah, nice. Right, so `K` becomes `P`...is an option. But I do also like your `VId` - amusingly it also addresses my complaint that "BB sounds like a variable" by changing the LHS (BB) rather than the RHS, breaking the idea that all-caps means variable ;-)....

The case of the original map from `str` to `Variable` also neatly identifies that a `str` isn't a variable, merely an identifier for one (and previously we were also using `str` where we now have `P`)